allennlp/common/params.py

```diff
@@ -354,6 +354,49 @@ def from_file(params_file: str, params_overrides: str = "") -> 'Params':
 
         return Params(param_dict)
 
+    def to_file(self, params_file: str, preference_orders: List[List[str]] = None):
+        with open(params_file, "w") as handle:
+            json.dump(self.as_ordered_dict(preference_orders), handle, indent=4)
+
+    def as_ordered_dict(self, preference_orders: List[List[str]] = None):
+        """
+        Returns Ordered Dict of Params from list of partial order preferences.
+
+        Parameters
+        ----------
+        preference_orders: List[List[str]], optional
+            ``preference_orders`` is list of partial preference orders. ["A", "B", "C"] means
+            "A" > "B" > "C". For multiple preference_orders first will be considered first.
+            Keys not found, will have last but alphabetical preference. Default Preferences:
+            ``[["dataset_reader", "iterator", "model", "train_data_path", "validation_data_path",
+                "test_data_path", "trainer", "vocabulary", "seeds"], ["type"]]``
+        """
+        params_dict = self.as_dict(quiet=True)
+        if not preference_orders:
+            preference_orders = []
+            preference_orders.append(["dataset_reader", "iterator", "model",
+                                      "train_data_path", "validation_data_path", "test_data_path",
+                                      "trainer", "vocabulary", "seeds"])
+            preference_orders.append(["type"])
+
+        def preference_ranks(key):
+            '''returns list of all preference scores'''
+            return [order.index(key) if key in order else len(order)
+                    for order in preference_orders]
+
+        def order_func(key):
+            '''returns list of all preference scores with default alphabetic score'''
```

Comment:
I think this would be easier to read if you combined these two functions into something like this:

```python
def order_func(key):
    # Makes a tuple to use for ordering.  The tuple is an index into each of the `preference_orders`,
    # followed by the key itself.  This gives us integer sorting if you have a key in one of the
    # `preference_orders`, followed by alphabetical ordering if not.
    order_tuple = [order.index(key) if key in order else len(order) for order in preference_orders]
    return order_tuple + [key]
```

Still exactly the same logic as what you have, but a bit easier on the reader because it's a single function with a good comment.

Reply:
Sure! Earlier I had two lambdas and so two ended up with functions.. but this is better!

Reply:
done 👍

Reply:
Done 👍 Thanks, this is better.